cluster/gce/gci/configure-helper.sh

```diff
@@ -1771,7 +1771,33 @@ function start-kube-proxy {
 # $5: pod name, which should be either etcd or etcd-events
 function prepare-etcd-manifest {
   local host_name=${ETCD_HOSTNAME:-$(hostname -s)}
-  local -r host_ip=$(python3 -c "import socket;print(socket.gethostbyname(\"${host_name}\"))")
+
+  local resolve_host_script_py='
+import socket
+import time
+import sys
+
+timeout_sec=300
+
+def resolve(host):
+  for attempt in range(timeout_sec):
+    try:
+      print(socket.gethostbyname(host))
+      break
+    except Exception as e:
+      sys.stderr.write("error: resolving host %s to IP failed: %s\n" % (host, e))
+      time.sleep(1)
+      continue
+
+'
+  if [[ -n "${MULTIMASTER:-}" ]]; then
+    # Multi-master should expose the peer server on the host level.
+    local -r host_ip=$(python3 -c "${resolve_host_script_py}"$'\n'"resolve(\"${host_name}\")")
+  else
+    # Single-master should expose the peer connection on the host level.
+    local host_ip=127.0.0.1
```

Comment:
Forgive my bash ignorance, but why we don't declare this one with -r?

Reply:
> -r	to make NAMEs readonly

We don't have consistent style of using it, but that's more 'defensive'

Reply:
Done